presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationPushDown.java

```diff
@@ -132,51 +139,95 @@ public void testOffsetPushDown() throws ExecutionException, InterruptedException
         assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), 2);
     }
 
-    @Test
+    @Test(retryAnalyzer = RetryAnalyzerForTimestampTest.class)
     public void testTimestampCreateTimeModePushDown() throws ExecutionException, InterruptedException
     {
-        Pair<String, String> timePair = createTimestampTestMessages(topicNameCreateTime);
+        RecordMessage recordMessage = createTimestampTestMessages(topicNameCreateTime);
         DistributedQueryRunner queryRunner = (DistributedQueryRunner) getQueryRunner();
         // ">= startTime" insure including index 2, "< endTime"  insure excluding index 4;
         String sql = String.format("SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s' and _timestamp < timestamp '%s'",
-                topicNameCreateTime, timePair.first(), timePair.second());
+                topicNameCreateTime, recordMessage.getStartTime(), recordMessage.getEndTime());
 
         // timestamp_upper_bound_force_push_down_enabled default as false.
         ResultWithQueryId<MaterializedResult> queryResult = queryRunner.executeWithQueryId(getSession(), sql);
-        assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), 998);
+        assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
+                .describedAs("with dump:\n%s", dumpResultAndData(topicNameCreateTime, recordMessage, queryRunner, sql, queryResult))
+                .isEqualTo(998);
```

Comment:
compute `dumpResultAndData(topicNameCreateTime, recordMessage, queryRunner, sql, queryResult)` on a variable and use for both assertions